### PR TITLE
Add a strategy to match free handles to connections.

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: runtime/strategies/match-free-handles-to-connections.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+export default class MatchFreeHandlesToConnections extends Strategy {
+  async generate(strategizer) {
+    let self = this;
+
+    let results = Recipe.over(this.getResults(strategizer), new class extends RecipeWalker {
+      onView(recipe, handle) {
+        if (handle.connections.length > 0)
+          return;
+
+        let matchingConnections = recipe.handleConnections.filter(connection => connection._handle == undefined);
```
Comment (dstockwell) [inline runtime/strategies/match-free-handles-to-connections.js]: There's a getter (still called `view`) to access `_handle`.

File: runtime/strategies/match-free-handles-to-connections.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+           
+        return matchingConnections.map(connection => {
+          return (recipe, handle) => {
+            let newConnection = recipe.updateToClone({connection}).connection;
+            newConnection.connectToView(handle);
```
Comment (dstockwell) [inline runtime/strategies/match-free-handles-to-connections.js]: Is this intentionally leaving the invalid recipes (where the type of handle does not match connection) for the strategizer to discard?

Comment (shans) [reply runtime/strategies/match-free-handles-to-connections.js]: Yeah. It's not ideal but I don't think we can do the awesome thing until we have a cleaner type system interface.

File: runtime/strategies/match-free-handles-to-connections.js (hunk 1 above)
Comment (shans) [reply runtime/strategies/match-free-handles-to-connections.js]: done.

**Pull request merged**